src/components/common/TmBalance.vue

```diff
@@ -298,6 +344,12 @@ export default {
           })
         return rewardsAccumulator
       }
+    },
+    preferredCurrency() {
+      return localStorage.getItem(`preferredCurrency`)
```

Comment:
this is badly implemented. it will always ask the localStorage. it should instead be a value on data which is set only when the currency is selected and on mounted

Reply:
after this anyone is free to merge

Reply:
Done!

And thank you for showing the correct implementation. I wasn't too convinced with my method either